### ISSUE TITLE
Globbed array outputs for a task not always expected length

### ISSUE THREAD

**Issue by jsotobroad** [broadinstitute/cromwell#843]
When running the workflow in #820 , the output arrays should each have 901 elements.  It seems that randomly the array outputs for some of the shards have less than 901 elements even though the output glob buckets have 901 elements in them.  This causes tasks further down in the workflow to fail.  This is not repeatable in terms of the same shards running into the issue but at this scale it seems to affect at least one shard per workflow.  351dd737-fe83-4e86-bc5b-19d4c7d2fbf7 - gotc dev

**Comment by jsotobroad** [comment]
task that consumes the array of arrays output in #820 

```
task MatrixRotation {
  Array[Array[String]] input_matrix

    command <<<
      python <<CODE
      import csv
      with open('${write_tsv(input_matrix)}') as tsv_in:
        input_matrix = [line.strip().split('\t') for line in tsv_in]
        final_matrix = [["" for x in range(len(input_matrix))] for y in range(len(input_matrix[0]))]
        for x in range(len(input_matrix)):
          for y in range(len(input_matrix[0])):
            final_matrix[y][x] = input_matrix[x][y]
      with open("output.tsv", "w") as tsv_out:
        # lineterminator is a workaround for a cromwell bug that doesnt allow for '\r\n' line endings which this outputs by default
        writer = csv.writer(tsv_out, delimiter='\t', lineterminator='\n')
        writer.writerows(final_matrix)
      CODE
      >>>

    runtime {
      docker: "python:2.7"
      memory: "1 GB"
      #preemptible: 3
    }

    output {
        Array[Array[String]] output_matrix = read_tsv("output.tsv")
    }
}
```
and this is how its used

```
  call MatrixRotation as RotateGVCF {
    input:
      input_matrix = SplitGvcfTouch.gvcf_list,
  }

  call MatrixRotation as RotateGVCFIndex {
    input:
      input_matrix = SplitGvcfTouch.gvcf_index_list,
  }
```

**Comment by cjllanwarne** [comment]
Shifting to twilight zone until we get a Google response.

**Comment by kcibul** [comment]
Tracked as Google Pipelines API Feature Request #28858407 and is in progress

**Comment by cjllanwarne** [comment]
We'll fix this in #1395
